### PR TITLE
Adds FeatureModelProvider to access FeatureModels

### PR DESCRIPTION
**Pull request by vulder**
The new FeatureModelProvider interface allows user of projects to access
FeatureModels that describe the Features and their relationships for a
project.

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- tests/provider/test_feature_model_provider.py  2
- varats-core/varats/provider/feature/feature_model_provider.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6273584&bid=20419891)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/provider/test_feature_model_provider.py  2
- varats-core/varats/provider/feature/feature_model_provider.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6273584&bid=20419891)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/provider/test_feature_model_provider.py  2
- varats-core/varats/provider/feature/feature_model_provider.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6273584&bid=20419891)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/provider/test_feature_model_provider.py  2
- varats-core/varats/provider/feature/feature_model_provider.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6273584&bid=20419891)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/provider/test_feature_model_provider.py  2
- varats-core/varats/provider/feature/feature_model_provider.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6273584&bid=20419891)

**Review by boehmseb — APPROVED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/provider/test_feature_model_provider.py  2
- varats-core/varats/provider/feature/feature_model_provider.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6273584&bid=20419891)

File: tests/provider/test_feature_model_provider.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        provider = FeatureModelProvider.create_provider_for_project(X264)
+        self.assertIsNotNone(provider)
+        if provider:
+            self.assertTrue(
+                str(provider.get_feature_model_path("dummy_revision")
+                   ).endswith("ConfigurableSystems/x264/FeatureModel.xml")
+            )
+
+    def test_false_feature_model_path_access(self):
+        """Checks look-up for none existens FeatureModels."""
```
Comment (ChristianKaltenecker) [inline tests/provider/test_feature_model_provider.py]:
```suggestion
        """Checks look-up for non-existent FeatureModels."""
```

**Review by ChristianKaltenecker — APPROVED** [review]

**Pull request merged**